youtube_dl/extractor/cjsw.py

```diff
@@ -0,0 +1,48 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+)
+
+
+class CJSWIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?cjsw\.com/program/\S+/(?P<id>[0-9]+)'
+    IE_NAME = 'cjsw'
+    _TEST = {
+        'url': 'http://cjsw.com/program/freshly-squeezed/episode/20170620',
+        'md5': 'cee14d40f1e9433632c56e3d14977120',
+        'info_dict': {
+            'id': '20170620',
+            'ext': 'mp3',
+            'title': 'Freshly Squeezed',
+            'description': 'Sled Island artists featured // Live session with Phi Pho, followed by a live session with Sinzere & The Late Nights! // Stay Fresh Y\'all!!',
+        }
+    }
+
+    def _real_extract(self, url):
+        episode_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, episode_id)
+
+        episode_controls = self._search_regex(r'<div[^>]+class=(["\'])episode-controls\1[^>]*>', webpage, 'episode_controls', fatal=False)
+        if not episode_controls:
+            raise ExtractorError('No streamable podcast', video_id=episode_id, expected=True)
```

Comment:
Some cjsw.com podcast pages don't have a stream to download ([this one for example](http://cjsw.com/program/bass-ackwards/episode/20170407/)). These lines are to search for the episode_controls div that is present when the page has a stream to download & to fail in an expected way if it doesn't. When the page doesn't have anything to download, it would be nice to make it clear that this isn't a bug with youtube_dl.

This does however add unnecessary complexity that could break the extractor when it should be able to work. I'm guessing this is why you want it out?

I saw that the [bandcamp extractor](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/bandcamp.py#L62) did this, but in a way that doesn't risk it breaking unnecessarily.